app/services/db/monthly_sp_auth_count/unique_monthly_auth_counts_by_partner.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+module Db
+  module MonthlySpAuthCount
+    module UniqueMonthlyAuthCountsByPartner
+      extend Reports::QueryHelpers
+
+      module_function
+
+      # @param [String] key label for billing (Partner requesting agency)
+      # @param [Array<String>] issuers issuers for the iaa
+      # @param [Date] start_date iaa start date
+      # @param [Date] end_date iaa end date
+      # @return [PG::Result, Array]
+      def call(key:, issuers:, start_date:, end_date:)
+        date_range = start_date...end_date
+
+        return [] if !date_range || issuers.blank?
+
+        # Query a month at a time, to keep query time/result size fairly reasonable
+        # The results are rows with [user_id, ial, auth_count, year_month]
+        months = Reports::MonthHelper.months(date_range)
+        queries = build_queries(issuers: issuers, months: months)
+
+        year_month_to_users_to_profile_age = Hash.new do |ym_h, ym_k|
+          ym_h[ym_k] = Hash.new { |user_h, user_k| }
+        end
+
+        queries.each do |query|
+          temp_copy = year_month_to_users_to_profile_age.deep_dup
+
+          with_retries(
+            max_tries: 3,
+            rescue: [
+              ActiveRecord::SerializationFailure,
+              PG::ConnectionBad,
+              PG::TRSerializationFailure,
+              PG::UnableToSend,
+            ],
+            handler: proc do
+              year_month_to_users_to_profile_age = temp_copy
+              ActiveRecord::Base.connection.reconnect!
+            end,
+          ) do
+            Reports::BaseReport.transaction_with_timeout do
+              ActiveRecord::Base.connection.execute(query).each do |row|
+                user_id = row['user_id']
+                year_month = row['year_month']
+                profile_age = row['profile_age'] < 0 ? nil : row['profile_age']
+
+                year_month_to_users_to_profile_age[year_month][user_id] = profile_age
+              end
+            end
+          end
+        end
+
+        rows = []
+        prev_seen_users = Set.new
+        year_months = year_month_to_users_to_profile_age.keys.sort
+
+        ial_to_year_month_to_users.each do |ial, year_month_to_users|
+          prev_seen_users = Set.new
+
+          year_months = year_month_to_users.keys.sort
+
+          year_months.each do |year_month|
+            year_month_users = year_month_to_users[year_month]
+
+            auth_count = year_month_users.count
+            unique_users = year_month_users.uniq.to_set
+        year_months.each do |year_month|
+          year_month_users = [year_month]
+          unique_users = year_month_users.uniq.to_set
+
+          new_unique_users = unique_users - prev_seen_users
+          new_unique_users_year1 = new_unique_users.count { |_user_id, profile_age| profile_age == 0 }
```

Comment:
indentation here is off, `year_months.each do |year_month|` is repeated, is this a merge artifact that needs to be cleaned up?